substrate/frame/system/src/offchain.rs

```diff
@@ -517,6 +517,29 @@ pub trait SignMessage<T: SigningTypes> {
 		TPayload: SignedPayload<T>;
 }
 
+/// Interface for creating a transaction for a call that will be authorized.
```

Comment:
what do we call transaction and what extrinsic? or are they interchangeable? I remember this comment from Gav, but from the Extrinisic Horizon text, they seems to be used interchangeably.
https://github.com/paritytech/substrate/pull/12092#discussion_r971801115

Reply:
IIUC, a block is made of extrinsics, before extrinsic horizon, there was:
* inherent
* transaction
  * unsigned
  * signed

extrinsic is what compose the block, transaction is a subtype of extrinsic which goes into the transaction pool, inherent don't go to the transaction pool.

With extrinsic horizon there is, extrinsics:
* bare: basically inherent and unsigned transaction (unsigned transaction are deprecated)
* general: they are transaction
* signed: they are transaction

All in all the way I understand is that inherent are not transaction, they don't go to the transaction pool, other stuff are transaction. And for now that is all, no third subtype of extrinsic.